api/v1alpha1/releaseplan_types.go

```diff
@@ -79,6 +97,39 @@ type ReleasePlan struct {
 	Status ReleasePlanStatus `json:"status,omitempty"`
 }
 
+// IsMatched checks whether the ReleasePlan is matched to a ReleasePlanAdmission.
+func (rp *ReleasePlan) IsMatched() bool {
+	return meta.IsStatusConditionTrue(rp.Status.Conditions, MatchedConditionType.String())
+}
+
+// MarkMatched marks the ReleasePlan as matched to a given ReleasePlanAdmission.
+func (rp *ReleasePlan) MarkMatched(releasePlanAdmission *ReleasePlanAdmission) {
+	rp.setMatchedStatus(releasePlanAdmission, metav1.ConditionTrue)
+}
+
+// MarkUnmatched marks the ReleasePlan as not matched to any ReleasePlanAdmission.
+func (rp *ReleasePlan) MarkUnmatched() {
+	if !rp.IsMatched() {
+		return
+	}
+
+	rp.setMatchedStatus(nil, metav1.ConditionFalse)
+}
+
+// setMatchedStatus sets the ReleasePlan Matched condition based on the passed releasePlanAdmission and status.
+func (rp *ReleasePlan) setMatchedStatus(releasePlanAdmission *ReleasePlanAdmission, status metav1.ConditionStatus) {
+	if releasePlanAdmission != nil {
+		rp.Status.ReleasePlanAdmission = MatchedReleasePlanAdmission{
```

Comment:
Just an idea... as both fields in MatchedReleasPlanAdmission are omittable, you could do something like:
```
rp.Status.RPA = MatchesReleasePlanAdmissiion{}

if rpa != nil {
    rp.Status.RPA.Name = ...
    rp.Status.RPA.Active = ...
}

conditions.SetCondition...
```

It's cleaner. Specially adding some empty lines :stuck_out_tongue_winking_eye: